File: imu_calib_matlab/tests/test_bundle_io.m
% test_bundle_io
% Validate manifest bundle export/load round-trip for the synthetic dataset.

clear;
clc;

projectRoot = fileparts(fileparts(mfilename('fullpath')));
addpath(genpath(fullfile(projectRoot, 'src')));

[data, ~] = load_example_data('forceRegenerate', true, 'saveToMat', false);
tmpDir = tempname;
mkdir(tmpDir);
cleanupObj = onCleanup(@() rmdir(tmpDir, 's')); %#ok<NASGU>

export_dataset_bundle(data, tmpDir);
loaded = load_csv_data(tmpDir);

assert(size(loaded.static.gyro, 1) == size(data.static.gyro, 1), ...
    'Static gyro sample count mismatch after bundle round-trip.');
assert(numel(loaded.accPoses) == numel(data.accPoses), ...
    'Accelerometer pose count mismatch after bundle round-trip.');
assert(numel(loaded.gyroRuns) == numel(data.gyroRuns), ...
    'Gyro run count mismatch after bundle round-trip.');
assert(isempty(loaded.accPoses(1).a_ref), ...
    'Default bundle export should omit legacy acc pose reference columns.');

fprintf('test_bundle_io passed.\n');

File: imu_calib_python/tests/test_bundle_io.py
import pandas as pd

from imu_calib.io.export_dataset_bundle import export_dataset_bundle
from imu_calib.io.load_csv_data import load_csv_data
from imu_calib.io.load_example_data import load_example_data


def test_bundle_round_trip(tmp_path):
    data, _, _ = load_example_data()
    export_info = export_dataset_bundle(data, tmp_path)
    loaded = load_csv_data(tmp_path)

    assert loaded.static.gyro.shape == data.static.gyro.shape
    assert len(loaded.acc_poses) == len(data.acc_poses)
    assert len(loaded.gyro_runs) == len(data.gyro_runs)
    assert export_info["manifest"].endswith("dataset_manifest.json")
    acc_df = pd.read_csv(tmp_path / "acc_poses.csv")
    assert "ref_x" not in acc_df.columns
    assert "ref_y" not in acc_df.columns
    assert "ref_z" not in acc_df.columns
